src/main/src/test/java/org/geoserver/catalog/impl/CatalogImplTest.java

```diff
@@ -1554,13 +1555,19 @@ private boolean layerHasSecurityRule(
 
         List<DataAccessRule> rules = dao.getRules();
         for (DataAccessRule rule : rules) {
-            if (rule.getRoot().equalsIgnoreCase(workspaceName)
-                    && rule.getLayer().equalsIgnoreCase(layerName)) return true;
+            if (rule.getRoot().equalsIgnoreCase(workspaceName) && sameLayer(layerName, rule))
+                return true;
         }
 
         return false;
     }
 
+    private boolean sameLayer(String layerName, DataAccessRule rule) {
```

Comment:
Name of method and what it actually does is surprising. It works, in the context of the calling point, because the root was already checked. Maybe move the entire test in the if inside the method? Then it would make more sense.

Reply:
removed method and inlined check. I think at some point it was called by more methods .